### PR TITLE
refactor: 💡Mirage changes to LDAP

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: ICU-10027

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10027)

## Description
Mirage changes to LDAP resource

### Screenshots (if appropriate):

Screenshot 2023-07-13 at 10 09 28

### CODE REVIEW

File: addons/api/mirage/generated/factories/auth-method.js [hunk 1]
```diff
@@ -43,6 +44,37 @@ export default Factory.extend({
           idp_ca_certs: ['certificate-1234', 'certificate-5678'],
         };
         break;
+      case TYPE_AUTH_METHOD_LDAP:
+        attrs = {
+          state: 'active-public',
+          urls: ['ldap://ldap.forumsys.com'],
```
Comment (lisbet-alvarez) [inline addons/api/mirage/generated/factories/auth-method.js:50]:
```suggestion
          urls: [`ldap://${faker.internet.domainName()}`],
```

File: addons/api/mirage/generated/factories/auth-method.js [hunk 2]
```diff
@@ -43,6 +44,37 @@ export default Factory.extend({
...
+          insecure_tls: false,
+          bind_dn: 'cn=read-only-admin,dc=example,dc=com',
+          bind_password: 'password',
+          bind_password_hmac: faker.random.alphaNumeric(50),
+          upn_domain: 'example.com',
+          discover_dn: false,
+          anon_group_search: false,
+          user_dn: 'dc=example,dc=com',
+          user_attr: 'uid',
+          user_filter: 'uid={username}',
```
[the remark this answers is not in the document]
Comment (ZedLi) [reply addons/api/mirage/generated/factories/auth-method.js]: Actually I was wondering how does it work? Does the backend use a template and the braces are correct syntax? It wasn't clear from the [documentation](https://developer.hashicorp.com/boundary/docs/concepts/domain-model/auth-methods#user_filter)

[the remark this answers is not in the document]
Comment (dvonanderson) [reply addons/api/mirage/generated/factories/auth-method.js]: @ZedLi good query got a thread opened with BE to see what is the correct default values.

**Review by DhariniJeeva — APPROVED** [review]
Do we need any domain config changes to support LDAP crud?

**Comment by dvonanderson** [comment]
> Do we need any domain config changes to support LDAP crud?

@DhariniJeeva I think the only thing I will do is make the `state=active-public`

**Pull request merged**